android-r.md

```diff
@@ -0,0 +1,95 @@
+
+#### Differences in the way it works across all Android versions
+
+The flow for that permission is the same across all android versions.
+
+### The background permission
+
+An app requires background location access if a feature within the app constantly shares location with other users or uses the Geofencing API.
+
+#### Requesting permission
+
+When your application requests the foreground permission, users won't see modal. He will be redirect to the settings screen which should look like this:
+
+[<img src="./assets/android-r/background_permissions_modal.png" width="300" />](./assets/asc-app-id/finding-app-id.png)
+
+That's why before that, you should inform the user why your application requires background permissions.
+
+**You can request the background permission only if the foreground permission was granted.**
+
+#### Differences in the way it works across all Android versions
+
+- On Android lower than Q, the background permission resolves to the foreground permission and the application won't redirect the user to the settings screen when asked.
+
+- On Android Q, the background permission will show up a similar system modal to the foreground permissions instead of redirecting to the settings screen.
+
+> If you want to know more about location permissions on Android, please check out this [article](https://medium.com/@ty2/understanding-permissions-for-background-location-on-android-11-and-below-bc3ad9be320a).
+
+## Scoped Storage
+
+It’s a concept of storing files, images, etc. separately, called Collections, which restrict the conventional access of the whole storage.
+
+The scoped storage was introduce in Android 10. However, we encouraged to opt-out from this feature in that version by using `requestLegacyExternalStorage` flag. This option is not available anymore when you're targeting Android 11.
+
+### What was changed?
+
+Some of the media library functions will ask the user for permissions before they perform any actions. You should be prepared that any of those functions may be rejected when the user doesn't grant the required permissions.
+
+[<img src="./assets/android-r/modifying_media_request.png" width="400" />](./assets/asc-app-id/finding-app-id.png)
+
+If your app created an album using an older SDK than 41 and you want to add more assets to this album, you need to migrate it to the new scoped directory. Otherwise, your app won't have access to the old album directory and the media library can't add new assets there. However, all other functions will work without problems. That's why you only need to migrate the old album if you want to add something to it.
+
+### Album migration
+
+To make the migration as easy as possible, we introduced two new functions into the media library:
+
+- `checkIfAlbumShouldBeMigratedAsync` which checks if the album should be migrated. In other words, if add operation, won't fail due to the lack of permissions.
```

Comment:
```suggestion
- `checkIfAlbumShouldBeMigratedAsync` which checks if the album should be migrated. In other words, it will check that new media can be added to the library — if not, then the album needs to be migrated.
```